### ISSUE TITLE
Add explicit hard negatives to manifold ranking training

### ISSUE THREAD

**Issue by t-yamsaki** [t-yamsaki/neembed#72]
## Goal

Extend the existing manifold multiple-negatives training path so users can provide explicit hard negatives without replacing the current aligned `(anchors, positives)` workflow.

Keep the v0.5 change small and backward-compatible.

## Scope

Extend `ManifoldMultipleNegativesRankingLoss` and the minimal trainer input path to support an explicit hard-negative batch form in addition to the existing aligned positive-pair form.

A reasonable public contract is:

```python
loss(anchors, positives)
loss(anchors, positives, negatives)
```

and corresponding trainer batches of either:

```text
(anchors, positives)
(anchors, positives, negatives)
```

The exact internal tensorization may be adjusted during implementation, but:
- positives remain the intended matches
- explicit negatives must contribute geodesic-distance logits
- existing in-batch negatives remain available
- Poincaré and Lorentz behavior must use the configured Geoopt manifold distance
- no negative-mining framework or dataset abstraction should be introduced

## Tests

Cover at least:
- existing two-sequence loss/trainer calls remain unchanged
- a three-sequence explicit-negative batch returns a finite scalar loss
- gradients reach ordinary model parameters
- explicit negatives influence the objective in the expected direction on a controlled toy example
- Poincaré and Lorentz paths are supported
- malformed or length-mismatched batches fail clearly
- normal CI remains network-independent

## Acceptance criteria

- [ ] existing `(anchors, positives)` training remains backward-compatible
- [ ] explicit hard negatives can be supplied directly
- [ ] geodesic distance remains the only similarity signal used by the manifold loss
- [ ] the trainer supports the new batch form without a callback/data-module framework
- [ ] gradients and losses remain finite for both supported geometries
- [ ] tests cover old and new batch contracts

## Out of scope

- automatic hard-negative mining
- ANN/vector-database integration
- distributed negatives
- cross-batch memory
- curriculum mining
- new optimizer abstractions

## Implementation principle

Add one explicit-negative path to the existing loss/trainer. Do not generalize the trainer into an arbitrary loss orchestration framework.
